### ISSUE TITLE
False positive "Unlisted dependencies" for resources with aliases

### ISSUE THREAD

**Issue by dartess** [webpro/knip#165]
First, a big thank you for this tool.

Some of my imports show up in the "Unlisted dependencies" section, which looks like a false positive.

1. import `scss` with aliases: `import 'shared/view/styles/index.scss';` or `import styles from 'features/ComponentName/ComponentName.module.scss';`. Relative imports works fine.

2. import `png` files: `import icon from 'shared/images/icon@3x.png';`. Relative imports also works fine.

**Comment by webpro** [comment]
Do you have repro repo? Or what is the output exactly?

**Comment by dartess** [comment]
My repo is private. Now I will make a separate demo with problems (I found another one).

**Comment by dartess** [comment]
@webpro repl: https://github.com/dartess/knip-bugs-repl/tree/main

The second error is that the knip analyzes the webpack's `entry`, but does not take into account `context`.:
```
Unresolved imports (1)
./index.tsx  webpack.config.ts
```
But `context` is `src` and `./src/index.tsx` exist.

I can open a separate issue if needed.

**Comment by webpro** [comment]
The issue with Webpack should fixed, I didn't know about the `context` property.

However, the initial issue isn't fixed; the issue is a deeper problem in Knip where TypeScript doesn't use take the `d.ts` files into account when resolving imports (maybe you can try to include them explicitly/manually). Then Knip has to decide what to with this:

```
shared/view/styles/index.scss
```

Is it a package in `node_modules` with a deeper path? Is it a local `.scss` file? When using relative paths, as you've seen, works best. Feel free to open a new issue for this, not sure when I can really dig into it, though.

(Let's open separate issues in the future to prevent this situation.)

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.16.2. See [Release 2.16.2](https://github.com/webpro/knip/releases/tag/2.16.2) for release notes._
